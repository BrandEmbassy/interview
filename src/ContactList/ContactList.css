.ContactList__empty {
  text-align: center;
  padding-top: 45%;
  color: #cccccc;
  transition-duration: 0.4s;
}