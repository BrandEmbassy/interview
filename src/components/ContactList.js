import { connect } from 'react-redux';
import List from './List';

const mapStateToProps = (state) => ({
  contacts: state.contacts,
});

const mapDispatchToProps = (dispatch) => ({
  onContactClick(id) {
  },
});

const ContactList = connect(
  mapStateToProps,
  mapDispatchToProps
)(List);

export default ContactList;
