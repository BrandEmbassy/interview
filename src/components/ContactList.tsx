/// <reference path="../../node_modules/typed-react/typed-react.d.ts" />
import TypedReact = require('typed-react')
/// <reference path="../../typings/react/react.d.ts" />
import React = require('react')
/// <reference path="../typings/react-redux/react-redux.d.ts" />
import { connect } from 'react-redux'
/// <reference path="../../typings/react-router/react-router.d.ts" />
import { Link } from 'react-router'
import { RouterProps } from '../utils/ReactUtils'

import ContactListItem = require('./ContactListItem')

import AppState = require('../model/AppState')
import Contact = require('../model/Contact')

interface ContactListProps extends RouterProps {
    contacts: Contact[]
}

class ContactList extends TypedReact.Component<ContactListProps, void> {

    private _contactRow(contact, index) {
        return <ContactListItem name={contact.fullName} id={contact.id} key={contact.id} />
    }

    public render() {

        // <div class="search">
        //     <input class="input" type="text" name="" value="" placeholder="Search ..." />
        // </div>
        // <div class="filter">
        //     <div class="filter__item filter__item--active">All</div>
        //     <div class="filter__item">A-Z</div>
        //     <div class="filter__item">Z-A</div>
        // </div>

        return (
            <div className="list">
                <div className="list__header">
                    <div className="heading">Contact List</div>
                </div>
                <div className="list__content">
                    {this.props.contacts.map(this._contactRow)}
                </div>
                <div className="list__footer">
                    <Link to='/contact/new'><div className="add-bttn"><span className="in">Add new contact</span></div></Link>
                </div>
            </div>
        )
    }   

}

function mapStateToProps(state: AppState) : ContactListProps {
    return {
        contacts: state.contacts
    }
}


export = connect(mapStateToProps)(TypedReact.createClass(ContactList))