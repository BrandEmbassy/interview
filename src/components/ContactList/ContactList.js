import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ContactListItem from '../ContactListItem/ContactListItem';

import { ContactType } from '../../types';
import './ContactList.css';

class ContactList extends Component {
  render() {
    return (
      <div className="ContactList">
        {this.props.contacts.map((contact, index, list) =>
          <ContactListItem
            key={contact.id}
            contact={contact}
            prevContact={index > 0 ? list[index - 1] : null}
            isReadOnly={this.props.isReadOnly}
            handleUpdate={this.props.handleUpdate}
            handleDelete={this.props.handleDelete}
          />
        )}

        {!this.props.contacts.length &&
          <div className="ContactList__empty">
            <h1>You still don't have any contact!</h1>
            <h2>You can fix it via the "Plus" button at the bottom-right :)</h2>
          </div>
        }
      </div>
    );
  }
}

ContactList.propTypes = {
  contacts: PropTypes.arrayOf(ContactType).isRequired
};

export default ContactList;
