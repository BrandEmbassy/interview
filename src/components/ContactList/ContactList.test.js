import React from 'react';

import {
  pucciMaurizioContact,
  rossiMarioContact,
} from '../../../.storybook/sample-contacts';

import ContactList from './ContactList';

test('should render the "no contacts" message', () => {
  const wrapper = shallow(
    <ContactList
      contacts={[]}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('should render one contact', () => {
  const wrapper = shallow(
    <ContactList
      contacts={[pucciMaurizioContact]}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('should render two contacts', () => {
  const wrapper = shallow(
    <ContactList
      contacts={[
        pucciMaurizioContact,
        rossiMarioContact,
      ]}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
