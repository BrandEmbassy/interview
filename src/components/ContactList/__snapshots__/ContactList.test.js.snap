// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one contact 1`] = `
<div
  className="ContactList"
>
  <ContactListItem
    contact={
      Object {
        "bio": "No comment!",
        "emails": Array [
          Object {
            "address": "maurizio.pucci@abc.com",
            "type": "lavoro",
          },
          Object {
            "address": "mao@xyz.io",
            "type": "privato",
          },
        ],
        "firstName": "Maurizio",
        "id": "100",
        "lastName": "Pucci",
        "phones": Array [
          Object {
            "number": "100 000 000",
            "type": "casa",
          },
        ],
      }
    }
    isNew={false}
    isReadOnly={false}
    key="100"
    prevContact={null}
  />
</div>
`;

exports[`should render the "no contacts" message 1`] = `
<div
  className="ContactList"
>
  <div
    className="ContactList__empty"
  >
    <h1>
      You still don't have any contact!
    </h1>
    <h2>
      You can fix it via the "Plus" button at the bottom-right :)
    </h2>
  </div>
</div>
`;

exports[`should render two contacts 1`] = `
<div
  className="ContactList"
>
  <ContactListItem
    contact={
      Object {
        "bio": "No comment!",
        "emails": Array [
          Object {
            "address": "maurizio.pucci@abc.com",
            "type": "lavoro",
          },
          Object {
            "address": "mao@xyz.io",
            "type": "privato",
          },
        ],
        "firstName": "Maurizio",
        "id": "100",
        "lastName": "Pucci",
        "phones": Array [
          Object {
            "number": "100 000 000",
            "type": "casa",
          },
        ],
      }
    }
    isNew={false}
    isReadOnly={false}
    key="100"
    prevContact={null}
  />
  <ContactListItem
    contact={
      Object {
        "bio": "Ciao!",
        "emails": Array [],
        "firstName": "Mario",
        "id": "300",
        "lastName": "Rossi",
        "phones": Array [],
      }
    }
    isNew={false}
    isReadOnly={false}
    key="300"
    prevContact={
      Object {
        "bio": "No comment!",
        "emails": Array [
          Object {
            "address": "maurizio.pucci@abc.com",
            "type": "lavoro",
          },
          Object {
            "address": "mao@xyz.io",
            "type": "privato",
          },
        ],
        "firstName": "Maurizio",
        "id": "100",
        "lastName": "Pucci",
        "phones": Array [
          Object {
            "number": "100 000 000",
            "type": "casa",
          },
        ],
      }
    }
  />
</div>
`;
